### PR TITLE
CVE-2017-5715 mitigation for Denver CPUs

### PR DESCRIPTION
**Pull request by vwadekar**

### CODE REVIEW

File: lib/cpus/aarch64/denver.S [hunk 1]
```diff
@@ -1,16 +1,152 @@
...
+	 * This register is accessed using MSR 3, 0, 15, 0, 6
+	 * and is only implemeted by Denver CPUs with MIDR
+	 * Primary Part number < 4.
+	 * -------------------------------------------------
+	 */
+	mrs	x0, midr_el1
+	lsr	x0, x0, #MIDR_PN_SHIFT
+	and	x0, x0, #MIDR_PN_MASK
+	tst	x0, #4
+	bne	1f
```
Comment (dp-arm) [inline lib/cpus/aarch64/denver.S]: you are doing a branch here before invalidating the BP.  Maybe you can use cmp/csel instead.

Comment (dp-arm) [reply lib/cpus/aarch64/denver.S]: If you cannot access this register on revisions that don't implement the mitigation, I guess you would have to move the id check before selecting the workaround vector table.

Comment (vwadekar) [reply lib/cpus/aarch64/denver.S]: @dp-arm 

The cpu_check_csv2 macro does something similar, so thought of keeping this logic here for Denver. Please let me know if my understanding is incorrect

**Comment by vwadekar** [comment]
@dp-arm 

Updated the PR to fix review comments.

**Comment by dp-arm** [comment]
jenkins: test this please

**Comment by dp-arm** [comment]
@vwadekar just realized, you need to use the declare_cpu_ops_workaround_cve_2017_5715
macro to set the check function which will report to higher layers whether the core is mitigated or not.

This is necessary to comply with the firmware mitigation specification. Check how it is done for Cortex A72 as an example.

**Comment by dp-arm** [comment]
@vwadekar you can send that as a separate patch of course.

**Comment by vwadekar** [comment]
@dp-arm 

Pushed PR #1388 to use the macro

**Pull request merged**